### ISSUE TITLE
Invalid config event received with zookeeper

### ISSUE THREAD

**Issue by xueyilong** [apache/shardingsphere#19079]
## Question

ShardingSphere-proxy version：5.1.1 、5.1.2
zookeeper version: 3.6.3

server.yaml
```
mode:
  type: Cluster
  #type: Memory
  repository:
    type: ZooKeeper
    props:
      namespace: governance_ds
      server-lists: 192.168.122.157:2181,192.168.122.221:2181,192.168.122.68:2181
      retryIntervalMilliseconds: 500
      timeToLiveSeconds: 60
      maxRetries: 3
      operationTimeoutMilliseconds: 500
  overwrite: false
```

When start the ShardingSphere-proxy，get the error logs,   want to get the reason, even if it doesn't seem to affect the use 
```
[INFO ] 2022-07-13 07:23:45.714 [main-SendThread(192.168.122.68:2181)] org.apache.zookeeper.ClientCnxn - Session establishment complete on server node3/192.168.122.68:2181, session id = 0x30014ae50ba0002, negotiated timeout = 40000
[INFO ] 2022-07-13 07:23:45.775 [main-EventThread] o.a.c.f.state.ConnectionStateManager - State change: CONNECTED
[INFO ] 2022-07-13 07:23:45.835 [main-EventThread] o.a.c.framework.imps.EnsembleTracker - New config event received: {server.2=192.168.122.157:2888:3888:participant, server.1=192.168.122.221:2888:3888:participant, server.3=192.168.122.68:2888:3888:participant, version=0}
[ERROR] 2022-07-13 07:23:45.839 [main-EventThread] o.a.c.framework.imps.EnsembleTracker - Invalid config event received: {server.2=192.168.122.157:2888:3888:participant, server.1=192.168.122.221:2888:3888:participant, server.3=192.168.122.68:2888:3888:participant, version=0}
[INFO ] 2022-07-13 07:23:45.839 [main-EventThread] o.a.c.framework.imps.EnsembleTracker - New config event received: {server.2=192.168.122.157:2888:3888:participant, server.1=192.168.122.221:2888:3888:participant, server.3=192.168.122.68:2888:3888:participant, version=0}
[ERROR] 2022-07-13 07:23:45.839 [main-EventThread] o.a.c.framework.imps.EnsembleTracker - Invalid config event received: {server.2=192.168.122.157:2888:3888:participant, server.1=192.168.122.221:2888:3888:participant, server.3=192.168.122.68:2888:3888:participant, version=0}
Thanks for using Atomikos! This installation is not registered yet.
```

**Comment by zjcnb** [comment]
@xueyilong Hi, Can you use ZooKeeper version `3.7.0` and try again? And it should not affect the use if there is this error message. I found related Information : https://stackoverflow.com/questions/62145737/error-main-eventthread-o-a-c-framework-imps-ensembletracker-invalid-config-eve

**Comment by xueyilong** [comment]
The error is still exists when use zk 3.7.0  and 3.7.1 ,and find another  error about java
```
[INFO ] 2022-07-13 14:40:40.076 [_finished_check_Worker-1-EventThread] o.a.c.f.state.ConnectionStateManager - State change: CONNECTED
[INFO ] 2022-07-13 14:40:40.081 [_finished_check_Worker-1-EventThread] o.a.c.framework.imps.EnsembleTracker - New config event received: {server.2=192.168.122.157:2888:3888:participant, server.1=192.168.122.221:2888:3888:participant, server.3=192.168.122.68:2888:3888:participant, version=0}
[ERROR] 2022-07-13 14:40:40.081 [_finished_check_Worker-1-EventThread] o.a.c.framework.imps.EnsembleTracker - Invalid config event received: {server.2=192.168.122.157:2888:3888:participant, server.1=192.168.122.221:2888:3888:participant, server.3=192.168.122.68:2888:3888:participant, version=0}
[INFO ] 2022-07-13 14:40:40.081 [_finished_check_Worker-1-EventThread] o.a.c.framework.imps.EnsembleTracker - New config event received: {server.2=192.168.122.157:2888:3888:participant, server.1=192.168.122.221:2888:3888:participant, server.3=192.168.122.68:2888:3888:participant, version=0}
[ERROR] 2022-07-13 14:40:40.081 [_finished_check_Worker-1-EventThread] o.a.c.framework.imps.EnsembleTracker - Invalid config event received: {server.2=192.168.122.157:2888:3888:participant, server.1=192.168.122.221:2888:3888:participant, server.3=192.168.122.68:2888:3888:participant, version=0}
Thread[JVMCI CompilerThread0,9,system]: Compilation of org.yaml.snakeyaml.constructor.Constructor$ConstructMapping.constructJavaBean2ndStep(MappingNode, Object) failed: java.lang.AssertionError: dependencies invalid Code installation failed: dependencies invalid
Failed dependency of type unique_concrete_method
  context = *org.yaml.snakeyaml.introspector.Property
  method  = {method} {0x00007f398081e6d8} 'getType' '()Ljava/lang/Class;' in 'org/yaml/snakeyaml/introspector/Property'
  witness = org.yaml.snakeyaml.introspector.PropertySubstitute

	at jdk.internal.vm.ci/jdk.vm.ci.hotspot.HotSpotCodeCacheProvider.installCode(HotSpotCodeCacheProvider.java:130)
	at jdk.internal.vm.compiler@11.0.15/org.graalvm.compiler.core.target.Backend.createInstalledCode(Backend.java:214)
	at jdk.internal.vm.compiler@11.0.15/org.graalvm.compiler.hotspot.CompilationTask.installMethod(CompilationTask.java:370)
	at jdk.internal.vm.compiler@11.0.15/org.graalvm.compiler.hotspot.CompilationTask$HotSpotCompilationWrapper.performCompilation(CompilationTask.java:191)
	at jdk.internal.vm.compiler@11.0.15/org.graalvm.compiler.hotspot.CompilationTask$HotSpotCompilationWrapper.performCompilation(CompilationTask.java:98)
	at jdk.internal.vm.compiler@11.0.15/org.graalvm.compiler.core.CompilationWrapper.run(CompilationWrapper.java:169)
	at jdk.internal.vm.compiler@11.0.15/org.graalvm.compiler.hotspot.CompilationTask.runCompilation(CompilationTask.java:330)
	at jdk.internal.vm.compiler@11.0.15/org.graalvm.compiler.hotspot.HotSpotGraalCompiler.compileMethod(HotSpotGraalCompiler.java:144)
	at jdk.internal.vm.compiler@11.0.15/org.graalvm.compiler.hotspot.HotSpotGraalCompiler.compileMethod(HotSpotGraalCompiler.java:111)
	at jdk.internal.vm.ci/jdk.vm.ci.hotspot.HotSpotJVMCIRuntime.compileMethod(HotSpotJVMCIRuntime.java:524)
To disable compilation failure notifications, set CompilationFailureAction to Silent (e.g., -Dgraal.CompilationFailureAction=Silent).
To print a message for a compilation failure without retrying the compilation, set CompilationFailureAction to Print (e.g., -Dgraal.CompilationFailureAction=Print).
Retrying compilation of org.yaml.snakeyaml.constructor.Constructor$ConstructMapping.constructJavaBean2ndStep(MappingNode, Object)
```

**Comment by linghengqian** [comment]
What if you don't use GraalVM? This error is not standard compared to OpemJDK.

**Comment by linghengqian** [comment]
- This issue was fixed at https://github.com/apache/shardingsphere/pull/21048 and should be closed.

**Comment by RaigorJiang** [comment]
> * This issue was fixed at [Update snakeyaml to 1.32 to fix yaml deserialization problem #21048](https://github.com/apache/shardingsphere/pull/21048) and should be closed.

Fixed, thank you!

**Comment by gonchik** [comment]
Great to found the fixed bug, which you meet in your prod ;)
